dockerfiles/pyne_dag_openmc

```diff
@@ -0,0 +1,142 @@
+    && pip install packaging \
+    progress pyyaml pytest openmc-plasma-source vtk tables
+
+# Install addition packages required for DAGMC
+# PyNE currently requires these versions of cython and numpy
+RUN pip install --upgrade numpy "cython<3.0" "numpy<2.0"
+# Clone and install EMBREE
+RUN mkdir -p $HOME/EMBREE && cd $HOME/EMBREE \
+    && git clone --single-branch -b ${EMBREE_TAG} --depth 1 ${EMBREE_REPO} \
+    && mkdir build && cd build \
+    && cmake ../embree \
+    -DCMAKE_INSTALL_PREFIX=${EMBREE_INSTALL_DIR} \
+    -DEMBREE_MAX_ISA=NONE \
+    -DEMBREE_ISA_SSE42=ON \
+    -DEMBREE_ISPC_SUPPORT=OFF \
+    && make 2>/dev/null -j${cores} install \
+    && rm -rf ${EMBREE_INSTALL_DIR}/build ${EMBREE_INSTALL_DIR}/embree
+
+# Clone and install MOAB
+RUN mkdir -p $HOME/MOAB && cd $HOME/MOAB \
+    && git clone  --single-branch -b ${MOAB_TAG} --depth 1 ${MOAB_REPO} \
+    && mkdir build && cd build \
+    && cmake ../moab -DENABLE_HDF5=ON \
+    -DENABLE_NETCDF=ON \
+    -DBUILD_SHARED_LIBS=ON \
+    -DENABLE_FORTRAN=OFF \
+    -DENABLE_BLASLAPACK=ON \
+    && make 2>/dev/null -j${cores} install \
+    && cmake ../moab \
+    -DENABLE_PYMOAB=ON \
+    -DBUILD_SHARED_LIBS=ON \
+    && make 2>/dev/null -j${cores} install \
+    && cd pymoab && bash install.sh \
+    && python setup.py install \
+    && python -c "import pymoab" \
+    && ldconfig
+
+# Clone and install Double-Down
+RUN mkdir -p $HOME/Double_down && cd $HOME/Double_down \
+    && git clone --single-branch -b ${DD_TAG} --depth 1 ${DD_REPO} \
+    && mkdir build && cd build \
+    && cmake ../double-down -DCMAKE_INSTALL_PREFIX=${DD_INSTALL_DIR} \
+    -DMOAB_DIR=/usr/local \
+    -DEMBREE_DIR=${EMBREE_INSTALL_DIR} \
+    && make 2>/dev/null -j${cores} install \
+    && rm -rf ${DD_INSTALL_DIR}/build ${DD_INSTALL_DIR}/double-down
+
+# Clone and install DAGMC
+RUN mkdir -p $HOME/DAGMC && cd $HOME/DAGMC \
+    && git clone --single-branch -b ${DAGMC_BRANCH} --depth 1 ${DAGMC_REPO} \
+    && mkdir build && cd build \
+    && cmake ../DAGMC -DBUILD_TALLY=ON \
+    -DCMAKE_INSTALL_PREFIX=${DAGMC_INSTALL_DIR} \
+    -DMOAB_DIR=/usr/local \
+    -DDOUBLE_DOWN=ON \
+    -DDOUBLE_DOWN_DIR=${DD_INSTALL_DIR} \
+    -DCMAKE_PREFIX_PATH=${DD_INSTALL_DIR}/lib \
+    -DBUILD_STATIC_LIBS=OFF \
+    && make 2>/dev/null -j${cores} install \
+    && rm -rf ${DAGMC_INSTALL_DIR}/DAGMC ${DAGMC_INSTALL_DIR}/build
+
+# clone and install openmc
+RUN mkdir -p ${HOME}/OpenMC && cd ${HOME}/OpenMC \
+    && git clone --shallow-submodules --recurse-submodules --single-branch -b ${OPENMC_BRANCH} --depth=1 ${OPENMC_REPO} \
+    && mkdir build && cd build \
+    && cmake ../openmc \
+    -DCMAKE_CXX_COMPILER=mpicxx \
+    -DOPENMC_USE_MPI=on \
+    -DHDF5_PREFER_PARALLEL=on \
+    -DOPENMC_USE_DAGMC=ON \
+    -DCMAKE_PREFIX_PATH=${DAGMC_INSTALL_DIR} \
+    && make 2>/dev/null -j${cores} install \
+    && cd ../openmc && pip install .[test,depletion-mpi] \
+    && python -c "import openmc"
```

Comment:
Can help with organization to have multiple build layers, separating dependencies from core software